stylix/home-manager-integration.nix

```diff
@@ -0,0 +1,80 @@
+homeManagerModule:
+{ lib, config, options, ... }:
+
+let
+  alwaysCopy = path: {
+    inherit path;
+    condition = lib.const true;
+  };
+
+  copyModules = builtins.map
+    (
+      { path, condition }:
+      { config, osConfig, ... }:
+      lib.mkIf (condition config)
```

Comment:
Although this `alwaysCopy` pattern with its `condition = lib.const true;` is ingenious, it makes it fairly hard to understand the code at first.

It might be simpler to remove the `alwaysCopy` function and replace the `alwaysCopy <PATH>` expressions in the list with `{ path = <PATH>; }`. In this case, `lib.mkIf` could check if `condition` exists and optionally evaluate it:

```nix
      lib.mkIf (x ? condition && x.condition config)
```

where `x` would be each item in the list.

This simplifies the common case and makes the rare case explicit.

Reply:
I wrote this as a function to make the code shorter, but I understand how it can be more complex to follow.

I'll change this to the format you described.

Reply:
Ended up using `lib.const true` as a default argument rather than your suggestion, since it would need to be

```nix
lib.mkIf (!(x ? condition) || (x.condition config))
```

Which IMO is a little confusing.